rules/lexer.l

```diff
@@ -55,6 +55,11 @@ L                       [a-zA-Z_]
 M                       [a-zA-Z_:]
 U                       [smhdwy]
 
+FLOAT                   [-+]?{D}*\.?{D}+{EXPONENT}?|[+-]?[iI][nN][fF]|[nN][aA][nN]
```

Comment:
Yeah, I left that out to keep the regex simpler, and all numbers are floats anyways. But now I simply changed the `{D}*\.?{D}+` part to `({D}*\.?{D}+|{D}+\.?{D}*)` and it supports that case too.